src/aioquic/quic/stream.py

```diff
@@ -207,6 +207,8 @@ def get_frame(
             if self._pending_eof:
                 # FIN only
                 self._pending_eof = False
+                if self._buffer_start == self._buffer_fin:
+                    self.is_finished = True
```

Comment:
I'm not sure I understand this bit: why does *sending* the FIN cause the sender to be finished?

Reply:
My understanding is that this is handled by QUIC's packet-based ACK/retransmission mechanism. We just need to make sure that we put out one packet with a FIN for retransmission to do its thing. Does that make sense?

Reply:
Put differently: QuicStreamSender has no retransmission logic for anything at all (neither data nor FINs) - that's handled at a different layer.

Reply:
That's not quite true: `QuicStreamSender` is informed of packet delivery or loss via the `on_data_delivery` method:

- on packet delivery it discards data
- on packet loss it reschedules data for transmission